libvips/colour/icc_transform.c

```diff
@@ -575,22 +570,24 @@ vips_image_expected_bands( VipsImage *image )
 		case VIPS_INTERPRETATION_RGB16:
 		case VIPS_INTERPRETATION_scRGB:
 		case VIPS_INTERPRETATION_HSV:
-			expected_bands = 3;
-			break;
+			/* The band count in the ICC profile must correspond to that of
+			 * the image, with a maximum of 3 bands allowed.
+			 */
+			return( VIPS_MIN( 3, image->Bands ) == profile_bands );
+
+		case VIPS_INTERPRETATION_CMYK:
+			/* CMYK images can only be imported if the ICC-profile has at
+			 * least 4 bands thereby blocking the usage of RGB profiles.
+			 */
+			return( profile_bands >= 4 );
 
 		case VIPS_INTERPRETATION_MULTIBAND:
 		case VIPS_INTERPRETATION_HISTOGRAM:
-		case VIPS_INTERPRETATION_CMYK:
 		case VIPS_INTERPRETATION_MATRIX:
 		case VIPS_INTERPRETATION_FOURIER:
 		default:
-			expected_bands = image->Bands;
-			break;
+			return( image->Bands == profile_bands );
```

Comment:
Maybe ">="? There can be more image bands than profile bands if there are alpha channels.

Reply:
Commit bec4942c6af0e629857da158c7e8657dfa0e1fff relaxes this check to `>=`.